src/getViewForSwagger2.ts

```diff
@@ -76,6 +77,65 @@ export function getViewForSwagger2(opts: CodeGenOptions): ViewData {
   };
 }
 
+function normalizeResponseDefinitions(swagger: any): any {
+  // ensure that the optional swagger.responses and swagger.definitions fields are present
+  swagger.responses = swagger.responses || {};
+  swagger.definitions = swagger.definitions || {};
+
+  // inject swagger.response defs into swagger.definitions
+  // prefixing them with "Response_" on name clashes with existing definitions
+  Object.entries<any>(swagger.responses).forEach(([name, def]) => {
+    if (!def.schema || def.schema.$ref) {
+      return;
+    }
+
+    const defName = (swagger.definitions[name] ? "Response_" : "") + name;
+    swagger.definitions[defName] = def.schema;
+    def.schema = { $ref: `#/definitions/${defName}` };
+  });
+
+  // inject inline response definitions into swagger.definitions
+  // the corresponding def name will be constructed like "Response_${opName}_${responseCode}"
+  // in order to avoid name clashes
+  getHttpMethodTuplesFromSwaggerPathsObject(swagger.paths).forEach(
+    ([path, httpVerb, op]) => {
+      const responses = op.responses;
+
+      Object.entries<any>(responses).forEach(([resCode, resDef]) => {
```

Comment:
This probably has a better type once we use the correct type for the function parameter?

Reply:
I think we won't need <any> here once we have the type in the comment above fixed.